pytket/pytket/utils/distribution.py

```diff
@@ -0,0 +1,268 @@
+# Copyright 2019-2022 Cambridge Quantum Computing
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import Counter, defaultdict
+from typing import cast, Any, Callable, DefaultDict, Dict, List, Set, Tuple, Union
```

Comment:
unused cast

Reply:
> the only major thing is I think it's worth using generic typing for this, I've demonstrated it here: #653 but I'm not actually sure if mypy is happy because I can't get it to run on my machine (my IDE type checking is happy at least)

That looks great, I'll check locally. I didn't know about `typing.Generic` before.

What do `OT` and `NT` stand for?

Reply:
> unused cast

Surprised pylint didn't pick this up...

Reply:
Fixed (by your commit)..

Reply:
> What do `OT` and `NT` stand for?

"Observable Type" and "New Type" I think, was just matching the examples in the generic docs